File: algorithms/typescript/bubble-sort.ts
function bubbleSort(array: number[])
{
	let sorted: boolean = false;

	while(sorted == false)
	{
		sorted = true;

		for(let i = 0; i < array.length - 1; i++)
		{
			if(array[i] > array[i + 1])
			{
				sorted = false;

				let temp = array[i];
				array[i] = array[i + 1];
				array[i + 1] = temp;
			}
		}
	}

	return array;
}

File: algorithms/go/bubble_sort.go
package algorithms

func BubbleSort(slice []int) {
	sorted := false

	if(len(slice) < 2) {
		return
	}

	for sorted == false {
		sorted = true

		for i := 0; i < len(slice) - 1; i++ {
			if slice[i] > slice[i + 1] {
				sorted = false

				temp1 := slice[i]
				temp2 := slice[i + 1]

				slice[i] = temp2
				slice[i + 1] = temp1
			}
		}
	}
}